### PR TITLE
Event cleanup

### PR DESCRIPTION
**Pull request by grobie**
Remove unused code, consolidate event enrichment code (less variables -> just fill the Event type attributes, less duplicated implementations), add option to filter watched events.

### CODE REVIEW

File: event.go [hunk 1]
```diff
@@ -116,201 +115,152 @@ func (s *Store) WatchEventRaw(listener chan *Event) error {
...
+	for _, t := range filter {
+		if e.Type == t {
+			return true
+		}
+	}
+	return false
+}
+
+func (e *Event) enrich(src cp.Event) error {
+	if !src.IsSet() {
```
Comment (xla) [inline event.go]: We should make sure that enrich has not effect on `EvUnknown`.

**Comment by xla** [comment]
:+1: :shipit: :octocat: :dango:

File: event.go (hunk 1 above)
Comment (grobie) [inline event.go]: Yeah, can add a test, but that event will just be ignored and source be set
to nil.

**Comment by matthiasr** [comment]
With my limited knowledge of the codebase, :+1: looks good. `TestEventFilter` could use a few more test cases though.

**Pull request merged**